File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        if(len(s) == 1):return 0
        
        sCount = [0]*26
        
        for i in range(len(s)):
            sCount[ord(s[i]) - ord('a')]+=1
        
        for i in range(len(s)):
            if(sCount[ord(s[i]) - ord('a')] == 1):
                return i
        
        return -1

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        map<char, int> m;
        map<char, int> ind;
        for(int i = 0; i < s.size(); i++){
            m[s[i]]++;
            ind[s[i]]=i;
        }
        for(int i = 0; i < s.size(); i++){
            if(m[s[i]] == 1){
                return ind[s[i]];
            }
        }
        return -1;
    }
};